src/engine/V3Eval.v3

```diff
@@ -965,6 +976,27 @@ def do_vv_v_x16(a: (u64, u64), b: (u64, u64), f: (u8, u8) -> u8) -> (u64, u64) {
 	}
 	return (low, high);
 }
+def do_v128_alltrue<T>(a: (u64, u64), bits: byte, view: u64 -> T, equal: (T, T) -> bool) -> bool {
+	for (shift = 0; shift < 64; shift += bits) {
+		var r1 = view(a.0 >> u6.view(shift));
+		var r2 = view(a.1 >> u6.view(shift));
+		if (equal(r1, T.!(0)) || equal(r2, T.!(0))) return false;
+	}
+	return true;
+}
+def do_v128_bitmask<T>(a: (u64, u64), bits: byte, bits_log2:byte, view: u64 -> T, less: (T, T) -> bool) -> i32 {
```

Comment:
If this returns an `u32`, then it can save a view in the caller. Or is there an important reason why it's signed?

Reply:
Thanks! Yep both the view and less are not necessary. I replaced the sign check with a direct bit manipulation.